library/core/src/ptr/non_null.rs

```diff
@@ -1416,6 +1417,8 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[unstable(feature = "pointer_is_aligned_to", issue = "96284")]
     #[rustc_const_unstable(feature = "const_pointer_is_aligned", issue = "104203")]
+    #[requires(align.is_power_of_two())]
```

Comment:
What happens if this condition is not satisfied? Since this is a safe function, we should verify that there is no UB even if this requirement is not met (panic is OK).

Reply:
I have added non_null_check_is_aligned_to_no_power_two for checking this! Let me know whether you have any suggestion. Can't wait to merge the PR!